core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -2887,66 +2889,123 @@ public void setTimestamp(long value) {
+                        throw CairoException.critical(ff.errno()).put("could not create [dir=").put(auxPath).put(']');
+                    }
+
+                    // backup table metadata files to TMP folder
+                    try {
+                        TableReaderMetadata metadata = reader.getMetadata();
+
+                        // _meta
+                        mem.smallFile(ff, auxPath.trimTo(tableRootLen).concat(TableUtils.META_FILE_NAME).$(), MemoryTag.MMAP_DEFAULT);
+                        metadata.dumpTo(mem);
+
+                        // create symbol maps
+                        auxPath.trimTo(tableRootLen).$();
+                        int symbolMapCount = 0;
+                        for (int i = 0, sz = metadata.getColumnCount(); i < sz; i++) {
+                            if (ColumnType.isSymbol(metadata.getColumnType(i))) {
+                                SymbolMapReader mapReader = reader.getSymbolMapReader(i);
+                                MapWriter.createSymbolMapFiles(
+                                        ff,
+                                        mem,
+                                        auxPath,
+                                        metadata.getColumnName(i),
+                                        COLUMN_NAME_TXN_NONE,
+                                        mapReader.getSymbolCapacity(),
+                                        mapReader.isCached());
+                                symbolMapCount++;
+                            }
+                        }
+
+                        // _txn
+                        mem.smallFile(ff, auxPath.trimTo(tableRootLen).concat(TableUtils.TXN_FILE_NAME).$(), MemoryTag.MMAP_DEFAULT);
+                        TableUtils.createTxn(mem, symbolMapCount, 0L, 0L, TableUtils.INITIAL_TXN, 0L, metadata.getStructureVersion(), 0L, 0L);
+
+                        // _cv
+                        mem.smallFile(ff, auxPath.trimTo(tableRootLen).concat(TableUtils.COLUMN_VERSION_FILE_NAME).$(), MemoryTag.MMAP_DEFAULT);
+                        TableUtils.createColumnVersionFile(mem);
+
+                        if (tableToken.isWal()) {
+                            // _name
+                            mem.smallFile(ff, auxPath.trimTo(tableRootLen).concat(TableUtils.TABLE_NAME_FILE).$(), MemoryTag.MMAP_DEFAULT);
+                            TableUtils.createTableNameFile(mem, tableToken.getTableName());
+
+                            // initialise txn_seq folder
+                            auxPath.trimTo(tableRootLen).concat(WalUtils.SEQ_DIR).slash$();
+                            if (ff.mkdirs(auxPath, configuration.getBackupMkDirMode()) != 0) {
+                                throw CairoException.critical(ff.errno()).put("Cannot create [path=").put(auxPath).put(']');
+                            }
+                            // _wal_index.d
+                            mem.smallFile(ff, auxPath.concat(WalUtils.WAL_INDEX_FILE_NAME).$(), MemoryTag.MMAP_DEFAULT);
```

Comment:
I don't think this is good enough.

if you change `TableBackupTest` to do inserts after backup

```Java
    @Test
    public void testMultipleTable() throws Exception {
        assertMemoryLeak(() -> {
            TableToken token1 = executeCreateTableStmt(testName.getMethodName());
            TableToken token2 = executeCreateTableStmt(token1.getTableName() + "_yip");
            executeSqlStmt("backup table " + token1.getTableName() + ", " + token2.getTableName());
            setFinalBackupPath();
            assertTables(token1);
            assertTables(token2);

            if (partitionBy != PartitionBy.NONE) {
                executeBackupSqlStmt("insert into " + token1.getTableName() + "(timestamp2) values('2023')");
                executeBackupSqlStmt("insert into " + token2.getTableName() + "(timestamp2) values('2023')");
            }
        });
    }

    private void executeBackupSqlStmt(String sql) throws SqlException {
        CairoEngine engine = mainEngine;
        SqlCompiler compiler = mainCompiler;
        SqlExecutionContext context = mainSqlExecutionContext;
        try {
            engine = new CairoEngine(new DefaultTestCairoConfiguration(finalBackupPath.toString()));
            context = TestUtils.createSqlExecutionCtx(engine);
            compiler = new SqlCompiler(engine);
            compiler.compile(sql, context).execute(null).await();
        } finally {
            Misc.free(engine);
            Misc.free(compiler);
            Misc.free(context);
        }
        drainWalQueue();
    }
```

It fails 

```
io.questdb.cairo.CairoException: [-100] Invalid metadata at fd=304. Timestamp index is outside of range, timestampIndex=0

	at io.questdb/io.questdb.cairo.CairoException.critical(CairoException.java:62)
	at io.questdb/io.questdb.cairo.TableUtils.validationException(TableUtils.java:1333)
	at io.questdb/io.questdb.cairo.TableUtils.getTimestampIndex(TableUtils.java:593)
	at io.questdb/io.questdb.cairo.wal.seq.SequencerMetadata.loadSequencerMetadata(SequencerMetadata.java:229)
	at io.questdb/io.questdb.cairo.wal.seq.SequencerMetadata.open(SequencerMetadata.java:148)
	at io.questdb/io.questdb.cairo.wal.seq.TableSequencerImpl.open(TableSequencerImpl.java:280)
	at io.questdb/io.questdb.cairo.wal.seq.TableSequencerAPI.openSequencerInstance(TableSequencerAPI.java:408)
	at io.questdb/io.questdb.std.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:967)
	at io.questdb/io.questdb.cairo.wal.seq.TableSequencerAPI.getTableSequencerEntry(TableSequencerAPI.java:381)
	at io.questdb/io.questdb.cairo.wal.seq.TableSequencerAPI.openSequencerLocked(TableSequencerAPI.java:414)
	at io.questdb/io.questdb.cairo.wal.seq.TableSequencerAPI.getTableMetadata(TableSequencerAPI.java:203)
	at io.questdb/io.questdb.cairo.pool.MetadataPool$SequencerMetadataTenant.<init>(MetadataPool.java:74)
	at io.questdb/io.questdb.cairo.pool.MetadataPool.newTenant(MetadataPool.java:47)
	at io.questdb/io.questdb.cairo.pool.MetadataPool.newTenant(MetadataPool.java:31)
	at io.questdb/io.questdb.cairo.pool.AbstractMultiTenantPool.get(AbstractMultiTenantPool.java:91)
	at io.questdb/io.questdb.cairo.CairoEngine.getMetadata(CairoEngine.java:371)
	at io.questdb/io.questdb.griffin.SqlCompiler.insert(SqlCompiler.java:1903)
```

Reply:
amended in 7b2fefb57ebbb043622a94fe0069d6807a3da1ee